docs/modules/ROOT/pages/documentation/microprofile/config/cloud/hashicorp.adoc

```diff
@@ -0,0 +1,78 @@
+# HashiCorp Secrets Config Source
+
+The HashiCorp Secrets Config Source takes configuration properties from HashiCorp Vault.
+
+## Prerequisite
+
+This documentation assumes the following:
+
+* You have successfully installed the HashiCorp Vault and have a running Vault server
+* You are familiar with most, if not all, of the HashiCorp Vault technology and terminology. If not please read through their documentation before you attempt to use HashiCorp Secrets Config Source
+* Your HashiCorp Vault is unsealed
+
+IMPORTANT: Hashicorp config source only supports the Vault KV secrets engine.
+
+[[configuration]]
+## Configuration in Payara Server
+
+You can configure HashiCorp Secrets Config Source either via the admin console or the _asadmin_ utility. You'll need a client token, the KV Secrets Engine API version, Vault's address represented as URL and port, the path where the kV secrets engine is enabled and the path where your secrets are stored.
+
+NOTE: By default, the KV secrets engine is enabled at the /secret path in Vault.
+
+### From the Admin Console
+
+To configure the `HASHICORP_VAULT_TOKEN` password alias from the admin console, refer to the xref:documentation/payara-server/password-aliases/password-alias-admin-console-commands.adoc[Password Aliases Admin Console configuration] guide.
+
+To configure the config source from the admin console, go to `Configs` -> `your-config` -> `MicroProfile` -> `Config` -> `HashiCorp Secrets`.
+
+image:microprofile/config/cloud/hashicorp/admin-console-config.png[Payara Server Administration Console configuration route]
+
+From here you can select the KV Secrets Engine API version you want to use, configure the vault address, the path where the KV secrets engine is enabled and the path where your secrets are stored. You can also decide whether to apply these changes dynamically or on the next server restart. If the config source is enabled or disabled dynamically it will take effect across the server immediately.
```

Comment:
Consistency:

```suggestion
From here you can select the KV Secrets Engine API version you want to use, configure the vault address, the path where the KV Secrets Engine is enabled and the path where your secrets are stored. You can also decide whether to apply these changes dynamically or on the next server restart. If the config source is enabled or disabled dynamically it will take effect across the server immediately.
```